apollo-integration/src/test/java/com/apollographql/android/impl/NormalizedCacheTestCase.java

```diff
@@ -0,0 +1,210 @@
+package com.apollographql.android.impl;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+
+import com.apollographql.android.api.graphql.Response;
+import com.apollographql.android.cache.normalized.CacheKeyResolver;
+import com.apollographql.android.impl.normalizer.EpisodeHeroName;
+import com.apollographql.android.impl.normalizer.HeroAndFriendsNames;
+import com.apollographql.android.impl.normalizer.HeroAndFriendsNamesWithIDForParentOnly;
+import com.apollographql.android.impl.normalizer.HeroAndFriendsNamesWithIDs;
+import com.apollographql.android.impl.normalizer.HeroAppearsIn;
+import com.apollographql.android.impl.normalizer.HeroParentTypeDependentField;
+import com.apollographql.android.impl.normalizer.HeroTypeDependentAliasedField;
+import com.apollographql.android.impl.normalizer.SameHeroTwice;
+import com.apollographql.android.impl.normalizer.type.Episode;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class NormalizedCacheTestCase {
+  private static final String NORMALIZER_TEST_PATH = "src/test/graphql/com/apollographql/android/impl/normalizer/";
+
+  private ApolloClient apolloClient;
+  private MockWebServer server;
+  private InMemoryCacheStore cacheStore;
+
+  @Before public void setUp() {
+    server = new MockWebServer();
+
+    OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+    cacheStore = new InMemoryCacheStore();
+
+    apolloClient = ApolloClient.builder()
+        .serverUrl(server.url("/"))
+        .okHttpClient(okHttpClient)
+        .normalizedCache(cacheStore, new CacheKeyResolver() {
+          @Nullable @Override public String resolve(Map<String, Object> jsonObject) {
```

Comment:
that how the we ported from iOS I guess. if null that means we are going to use path as a key.
We can change `CacheKeyResolver` to return not String but some class like `CacheKey`. And null replace with `CacheKey.EMPTY` smth like this